sdk/storage/Azure.Storage.Blobs/tests/BlockBlobClientTests.cs

```diff
@@ -1980,9 +1987,12 @@ public async Task UploadAsync_WithUnreliableConnection()
                 new IOException("Simulated stream fault"),
                 () => timesFaulted++))
             {
-                await blobFaulty.UploadAsync(stream, null, metadata, null);
+                await blobFaulty.UploadAsync(stream, null, metadata, null, progressHandler: progressHandler);
 
-                Assert.AreEqual(stream.Length, stream.Position);
+                await WaitForProgressAsync(progressList, data.LongLength);
+                Assert.IsTrue(progressList.Count > 1, "Too few progress received");
+                // Changing from Assert.AreEqual because these don't always update fast enough
+                Assert.GreaterOrEqual(data.LongLength, progressList.LastOrDefault(), "Final progress has unexpected value");
```

Comment:
shouldn't we add items in progress list and then compare with equality?

Reply:
This is a git reversion. Also I'm unsure of what you mean by adding items in the progress list. We report total transferred bytes, not bytes transferred since last report.

Reply:
gotcha. I didn't realize this is cumulative.